fixes issue 4691

bitFiedler opened the pull request:
fixes the behavior in issue 4691
https://github.com/nlohmann/json/issues/4691

Commit:
bugfix removeprefix not available in python version < 3.9

there was a problem in my cLion gdb that comes with python version 3.8.
the pretty printer script do not work because the removeprefix function is added in version 3.9.
therefore we need a fix for that.

Signed-off-by: Matthias Popp <mpopp@arri.at>

nlohmann commented on tools/gdb_pretty_printer/nlohmann-json.py:
```diff
@@ -15,18 +17,29 @@ def to_string(self):
+                try:
+                    key_val = None
+                    # python < 3.9 do not have removeprefix
+                    if py_ver.major == 3 and py_ver.minor <= 8 or py_ver.major < 3:
```
I think this check makes the code unnecessarily complex. Why not just implementing the version without `removeprefix`?

bitFiedler replied:
no problem, I reworked my commits

Commit:
fix indentation, use the same indentation size at in class JsonValuePrinter

the indentation size was a mix up, we should use always the same indentation size of 4

Signed-off-by: Matthias Popp <mpopp@arri.at>

Commit:
use prefix as variable

easier handling or easier to change

Signed-off-by: Matthias Popp <mpopp@arri.at>

Commit:
add variables for easier access

Signed-off-by: Matthias Popp <mpopp@arri.at>

nlohmann commented on tools/gdb_pretty_printer/nlohmann-json.py:
```diff
@@ -15,18 +15,21 @@ def to_string(self):
+        name = m.group('name')
+        if name and name.startswith('basic_json<') and name.endswith('>'):
+            m_data = val["m_data"]
+            m_type = m_data['m_type']
```
Why did you introduce these variables?

gregmarr replied:
Looks to be to eliminate duplicate accesses.  However, `m_type` is only accessed once in the normal case.  The second access is when an exception has been thrown.  `m_data` is accessed twice, or three times if the exceptional case is made to be `m_data["m_type"]`.

While you're here, probably worth changing `"m_data"` to `'m_data'` to match the rest of the function.

bitFiedler replied:
Yes removing duplicates, but if you look in the history some gdb fixes fall in that access. That means if the interface change in the the future the changes should be minimal. I will provide a rebased fix with the single quotes

nlohmann reviewed (changes requested):
Please check the DCO warning: https://github.com/nlohmann/json/pull/4693/checks?check_run_id=38900328997

nlohmann reviewed (approved):
Looks good to me.
